big_tests/tests/mod_global_distrib_SUITE.erl

```diff
@@ -563,96 +584,113 @@ test_pm_with_disconnection_on_other_server(Config) ->
-    EveSpec = ?config(evespec_mim, Config),
-    try
-        escalus:fresh_story(
-          Config, [{alice, 1}],
-          fun(Alice) ->
+    EveSpec = ?config(evespec_reg, Config),
+    EveSpec2 = ?config(evespec_mim, Config),
+    escalus:fresh_story(
+      Config, [{alice, 1}],
+      fun(Alice) ->
               Eve = connect_from_spec(EveSpec, Config),
 
               escalus_client:send(Eve, escalus_stanza:chat_to(Alice, <<"Hi from Asia!">>)),
-              escalus_connection:stop(Eve),
+
+              %% Stop connection and wait for process to die
+              EveNode = ct:get_config({hosts, reg, node}),
+              mongoose_helper:logout_user(Config, Eve, EveNode),
 
               FromEve = escalus_client:wait_for_stanza(Alice),
 
               escalus_client:send(Alice, chat_with_seqnum(Eve, <<"Hi from Europe1!">>)),
-              NewEve = connect_from_spec([{port, 5222} | EveSpec], Config),
-              %% TODO: This basically makes it way less likely to fail. Fix some day.
-              timer:sleep(500),
+              NewEve = connect_from_spec(EveSpec2, Config),
 
               escalus_client:send(Alice, chat_with_seqnum(Eve, <<"Hi again from Europe1!">>)),
               escalus_client:send(NewEve, escalus_stanza:chat_to(Alice, <<"Hi again from Asia!">>)),
 
               FirstFromAlice = escalus_client:wait_for_stanza(NewEve),
               AgainFromEve = escalus_client:wait_for_stanza(Alice),
-              SecondFromAlice = try escalus_client:wait_for_stanza(NewEve)
-                                catch
-                                    error:timeout_when_waiting_for_stanza ->
-                                        ct:log("Eve had received ~p~nAnd Alice had received ~p~n",
-                                               [FirstFromAlice, AgainFromEve]),
-                                        throw(global_distrib_issue)
-                                end,
+              SecondFromAlice = escalus_client:wait_for_stanza(NewEve),
 
               [FromAlice, AgainFromAlice] = order_by_seqnum([FirstFromAlice, SecondFromAlice]),
 
               escalus:assert(is_chat_message, [<<"Hi from Europe1!">>], FromAlice),
               escalus:assert(is_chat_message, [<<"Hi from Asia!">>], FromEve),
               escalus:assert(is_chat_message, [<<"Hi again from Europe1!">>], AgainFromAlice),
               escalus:assert(is_chat_message, [<<"Hi again from Asia!">>], AgainFromEve)
-          end)
-    catch
-        throw:global_distrib_issue -> {skip, "Test skipped, failure is known and not relevant"}
-    end.
+          end).
 
-test_pm_with_ungraceful_reconnection_to_different_server(Config0) ->
 %% This tests the feature which has not been implemented (yet?) by mod_global_distrib
 %% It is susceptible to a race condition, however it is very unlikely to occur
 %% See PR #2392
+test_pm_with_ungraceful_reconnection_to_different_server(Config) ->
```

Comment:
This is pretty much randomly choosing which of the next two tests will happen first, right? Don't think we need it then, might continue getting randomly red and it adds nothing :)